### PR TITLE
Integration test to verify that ADAL(U) can read RT written by MSAL(U)

### PR DESCRIPTION
**Pull request by jasoncoolmax**
Add test case to verify that ADAL(U) can read RT written by MSAL(U) 

Besides, found a bug that we didn't add cache delete event for `MSIDLegacyTokenCacheAccessor`, PR here: https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/115

### CODE REVIEW

File: ADAL/tests/XCTestCase+TestHelperMethods.h [hunk 1]
```diff
@@ -161,6 +163,10 @@
 - (MSIDRefreshToken *)adCreateRefreshToken;
 - (MSIDLegacySingleResourceToken *)adCreateLegacySingleResourceToken;
 
+- (MSIDRequestParameters *)adCreateV2DefaultParams;
```
Comment (oldalton) [inline ADAL/tests/XCTestCase+TestHelperMethods.h]: nit: don't we have this functionality in MSID tests?

Comment (jasoncoolmax) [reply ADAL/tests/XCTestCase+TestHelperMethods.h]: Yes, we do. Let me try to include that MSID test utility class to AD test :)

Comment (jasoncoolmax) [reply ADAL/tests/XCTestCase+TestHelperMethods.h]: I tried but seems there is no way to include test utility file in common core project to AD test :P

Comment (oldalton) [reply ADAL/tests/XCTestCase+TestHelperMethods.h]: Ok, let's leave it that way for now :) I suggest we merge this one.

**Pull request merged**